docs/source/dataset.rst

```diff
@@ -0,0 +1,232 @@
+   :header-rows: 1
+
+   * - raster path 
+     - ...
+   * - my_dir/my_multiband_geofile.tif
+     - ...
+   
+Multi-band raster files with more bands or different order than needed 
+======================================================================
+
+Same as above, but the expected order or subset of bands must be set in the 
+"*bands*" parameter of :ref:`dataset config <yamlparameters>`. For example:
+
+.. code-block:: yaml
+
+    dataset:
+        [...]
+        bands: [3, 2, 1]
+
+Here, if the original multi-band raster had "*BGR*" bands, geo-deep-learning will reorder these bands
+to "*RGB*" order. Also, band selection and reordering can be performed simultaneously. In this example,
+if the source raster contained "*BGRN*" bands, the "*N*" band would be removed and the resulting 
+raster would still be as "*RGB*".
+
+The ``bands`` parameter is set in the 
+`dataset config <https://github.com/NRCan/geo-deep-learning/blob/develop/config/dataset/test_ci_segmentation_multiclass.yaml>`_.
+
+.. note:: 
+
+    It's the user's responsibility to know which bands correspond to each index. When dealing with 
+    multi-band source imagery, Geo-Deep-Learning doesn't "*know*" which bands are present in the file 
+    and in which order. Good to know, Geo-Deep-Learning following the GDAL convention, bands are 
+    indexed from 1 
+    (`docs <https://rasterio.readthedocs.io/en/latest/quickstart.html#reading-raster-data>`_).
+
+
+Single-band raster files, identified with a common string pattern
+=================================================================
+
+A single line in the csv can refer to multiple single-band rasters, by using a string pattern.
+The "*band specific*" string in the file name must be in a 
+`hydra-like interpolation format <https://hydra.cc/docs/1.0/advanced/override_grammar/basic/#primitives>`_,
+with ``${...}`` notation. Geo-deep-learning will locate a list of single-band rasters during 
+execution using two informations:
+1. the "*bands*" parameter of :ref:`dataset config <yamlparameters>`.
+2. the input csv's path, if it contains the ``${dataset.bands}`` string.
+
+For example:
+
+1. Bands in `dataset configuration <https://github.com/NRCan/geo-deep-learning/blob/develop/config/dataset/test_ci_segmentation_binary.yaml>`_.:
+
+    .. code-block:: yaml
+
+        dataset:
+            [...]
+            bands: ["red", "green", "blue"]
+
+2. `Input csv <https://github.com/NRCan/geo-deep-learning/blob/develop/tests/tiling/tiling_segmentation_binary_ci.csv>`_:
+
+    .. list-table:: 
+        :header-rows: 1
+
+        * - raster path 
+          - ...
+        * - my_dir/my_singleband_geofile_band_${dataset.bands}.tif
+          - ...
+
+During execution, this would result in using, **in the same order as bands appear in dataset config**, 
+the following files:
+
+.. code-block:: shell
+
+    my_dir/my_singleband_geofile_band_R.tif
+    my_dir/my_singleband_geofile_band_G.tif
+    my_dir/my_singleband_geofile_band_B.tif

```

Comment:
In the above example, the `${dataset.bands}` parameter is set to red, green and blue, not r, g and b.